doc/design/docker-cp.md

```diff
@@ -119,15 +119,17 @@ The personality code should be fairly straight forward, the goal is to call the
  
  2) depending on the endpoint hit depends on the to/from scenario either way in an *off* scenario we would first get the container object and confirm that it is off. 
  
- 3) Once confirmed that the container is off we will need to immediately block starts(and even wait if one is triggered between the status check and requesting the lockdown). In addition to this, we must block other `cp` operations until the current one is complete. 
+ 3) Once confirmed that the container is off. We will make the request to mount for the import/export operation. 
+ 
+ 4) Then we will mount the r/w layer at a temp directory location and block all start operations on this container as we have the r/w already mounted to different compute device. In the even of a `copyFrom` scenario the disk should be mounted as `non-persistent`
+ 
+ 5) Begin streaming the tar file to the mounted disk at the assembled path. 
  
- 4) Then we will mount the r/w layer at a temp directory location. and assemble that path along with the target copy path.
+ 6) We will need to untar this in place while checking for existing files and behaving according to the overwrite dir flag. We will need this to be in place since we could be operating with limited space, we want to do this in a fashion that we have as little storage overhead during unpacking as possible. Checking flags for `untar` may help significantly with this. Otherwise we may need to read the tar headers themself before attempting the write and determining if we have the size to do this.  
  
- 5) Begin streaming the tar file to the mounted disk. 
  
- 6) We will need to untar this in place while checking for existing files and behaving according to the overwrite dir flag. We will need this to be in place since we could be operating with limited space, we want to do this in a fashion that we have as little storage overhead during unpacking as possible. Checking flags for `untar` may help significantly with this. Otherwise we may need to read the tar headers themself before attempting the write and determining if we have the size to do this. 
  
- 7) Upon successfully copying the data we detach the disk.
+ 7) Upon successfully copying the data we will adjust the permissions to be appropriately aligned with our decisions. Then detach the disk.
```

Comment:
We should be able to set the permissions during the extraction rather than doing it after the fact (which is racy in an error scenario). If need be we can update the archive/tar package to add this capability.